app/hotels/src/singleHotel/HotelDetailScreen.js

```diff
@@ -2,27 +2,35 @@
 
 import * as React from 'react';
 import { ScrollView } from 'react-native';
-import { Layout } from '@kiwicom/react-native-app-common';
+import { GeneralError, Layout } from '@kiwicom/react-native-app-common';
 
 import HeaderContainer from './header/HeaderContainer';
 import LocationContainer from './location/LocationContainer';
 import DescriptionContainer from './description/DescriptionContainer';
 
 import type { Image } from '../gallery/GalleryGrid';
-import type { singleHotelQueryResponse } from './__generated__/singleHotelQuery.graphql';
 
 export type Props = {|
   openGallery: (hotelName: string, images: Image[]) => void,
-  hotel: singleHotelQueryResponse,
+  availableHotel: any,
 |};
 
-export default function HotelDetailScreen({ openGallery, hotel }: Props) {
+export default function HotelDetailScreen({
+  openGallery,
+  availableHotel,
+}: Props) {
+  if (!availableHotel) {
+    return <GeneralError errorMessage="Hotel not found" />;
```

Comment:
I am trying to understand you but this is really confusing - you explained to me that this is not an error in your opinion and I merged different solution yesterday. Now you consider it an error and you are OK with this completely different semantics. Can you please explain me one more time what do you think about this so I can understand it better? 🙂

Reply:
In this case I think it is an error. User should never see this page, because he goes to the detail always straight from a search or a map. I've seen it only because of an error in GraphQL (I sent a hotel ID, but the server expected a hotel availability ID). Maybe the error message is a little misleading.

On the other hand, if you search for a hotel in [a location where there are no hotels](http://hotels.kiwi.com/searchresults.en-gb.html?lang=en-gb&sid=d6f39850478096fc63133f8249941def&sb=1&src=searchresults&src_elem=sb&error_url=http%3A%2F%2Fhotels.kiwi.com%2Fsearchresults.en-gb.html%3Faid%3D1165797%3Bsid%3Dd6f39850478096fc63133f8249941def%3Bcheckin_month%3D1%3Bcheckin_monthday%3D6%3Bcheckin_year%3D2018%3Bcheckout_month%3D1%3Bcheckout_monthday%3D7%3Bcheckout_year%3D2018%3Bclass_interval%3D1%3Bdtdisc%3D0%3Bfrom_sf%3D1%3Bgroup_adults%3D2%3Bgroup_children%3D0%3Binac%3D0%3Bindex_postcard%3D0%3Blabel_click%3Dundef%3Bmap%3D1%3Bno_rooms%3D1%3Boffset%3D0%3Boos_flag%3D0%3Bplace_id%3DChIJoezURc-WeK4RnnwrvdBqN-k%3Bplace_id_lat%3D-66.28206879999999%3Bplace_id_lon%3D110.52766309999993%3Bpostcard%3D0%3Braw_dest_type%3Dairport%3Broom1%3DA%252CA%3Bsb_price_type%3Dtotal%3Bsearch_selected%3D1%3Bsrc%3Dindex%3Bsrc_elem%3Dsb%3Bss%3DAntarctica%2520International%2520Airport%252C%2520Main%2520Road%252C%2520Casey%2520Station%252C%2520Antarctica%3Bss_all%3D0%3Bss_raw%3Dantarctida%3Bssb%3Dempty%3Bsshis%3D0%3Bssne_untouched%3DBangkok%26%3B&ss=Antarctica&ssne=Antarctica+International+Airport%2C+Main+Road%2C+Casey+Station%2C+Antarctica&ssne_untouched=Antarctica+International+Airport%2C+Main+Road%2C+Casey+Station%2C+Antarctica&checkin_monthday=6&checkin_month=1&checkin_year=2018&checkout_monthday=7&checkout_month=1&checkout_year=2018&group_adults=2&group_children=0&no_rooms=1&highlighted_hotels=&from_sf=1&search_pageview_id=612468035205024e&ac_position=0&ss_short=&dest_id=ChIJS3WQM3uWuaQRdSAPdB--Um4&dest_type=country&place_id=ChIJS3WQM3uWuaQRdSAPdB--Um4&place_id_lat=-75.250973&place_id_lon=-0.07138899999995374&place_types=country%2Cpolitical&search_pageview_id=612468035205024e&search_selected=true&ss_raw=Antarctica), the empty list is not an error. Similarly if you google a word that doesn't exist, Google doesn't respond with a 404 status.